src/Faithlife.Data/DbConnectorCommand.cs

```diff
@@ -264,10 +270,40 @@ public async ValueTask<DbConnectorResultSets> QueryMultipleAsync(CancellationTok
 			return new DbConnectorResultSets(command, await methods.ExecuteReaderAsync(CachedCommand.Unwrap(command), cancellationToken).ConfigureAwait(false), methods);
 		}
 
+		/// <summary>
+		/// Sets the command's timeout.
+		/// <remarks>Throws <c><see cref="ArgumentOutOfRangeException"/></c> if <c>timeSpan</c> is not positive or <c><see cref="Timeout.InfiniteTimeSpan"/></c>.</remarks>
+		/// </summary>
+		public DbConnectorCommand Timeout(TimeSpan timeSpan)
+		{
+			int? timeout;
+			if (timeSpan == System.Threading.Timeout.InfiniteTimeSpan)
+				timeout = 0;
+			else if (timeSpan <= TimeSpan.Zero)
+				throw new ArgumentOutOfRangeException(nameof(timeSpan), "Must be positive or 'Timeout.InfiniteTimeSpan'.");
+			else
+				timeout = (int?)Math.Ceiling(timeSpan.TotalSeconds);
+
+			return new DbConnectorCommand(Connector, Text, Parameters, timeout, IsCached);
+		}
+
+		/// <summary>
+		/// Sets the command's timeout.
+		/// <param name="timeout">The desired timeout of the command, in seconds. A value of <c>0</c> disables timeout.</param>
+		/// <remarks>Throws <c><see cref="ArgumentOutOfRangeException"/></c> if <c>timeout</c> is negative.</remarks>
+		/// </summary>
+		public DbConnectorCommand Timeout(int timeout)
```

Comment:
If we kept this, it would probably be named `WithTimeoutSeconds`. But using a `TimeSpan` is less confusing (mostly because zero means no timeout), so I'd like to stick with just the `TimeSpan` method for now.